module/app/app.go

```diff
@@ -906,18 +919,35 @@ func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino
 	return paramsKeeper
 }
 
+// Registers handlers for all our upgrades
 func (app *Gravity) registerUpgradeHandlers() {
-	app.upgradeKeeper.SetUpgradeHandler("v2", func(ctx sdk.Context, plan upgradetypes.Plan, _ module.VersionMap) (module.VersionMap, error) {
-		// 1st-time running in-store migrations, using last version as fromVersion to
-		// avoid running InitGenesis.
-		fromVM := make(map[string]uint64)
-		for moduleName, module := range app.mm.Modules {
-			fromVM[moduleName] = module.ConsensusVersion()
-		}
-
-		// set gravity version to 1 in order to execute migration scripts from 1 to 2
-		fromVM[gravitytypes.StoreKey] = 1
+	upgrades.RegisterUpgradeHandlers(
+		app.mm, app.configurator, app.accountKeeper, app.bankKeeper, app.bech32IbcKeeper, app.distrKeeper,
+		app.mintKeeper, app.stakingKeeper, app.upgradeKeeper,
+	)
+}
 
-		return app.mm.RunMigrations(ctx, app.configurator, fromVM)
-	})
+// Sets up the StoreLoader for new, deleted, or renamed modules
+func (app *Gravity) registerStoreLoaders() {
+	// Read the upgrade height and name from previous execution
+	upgradeInfo, err := app.upgradeKeeper.ReadUpgradeInfoFromDisk()
+	if err != nil {
+		panic(fmt.Sprintf("failed to read upgrade info from disk %s", err))
+	}
+
+	// v1->v2 STORE LOADER SETUP
+	// Register the new v2 modules and the special StoreLoader to add them
+	if upgradeInfo.Name == v2.V1ToV2PlanName {
+		if !app.upgradeKeeper.IsSkipHeight(upgradeInfo.Height) { // Recognized the plan, need to skip this one though
+			storeUpgrades := storetypes.StoreUpgrades{
+				Added: []string{bech32ibctypes.ModuleName}, // We are adding these modules
+				// Check upgrade docs to see which type of store loader is necessary for deletes/renames
```

Comment:
I think this can be removed now?

Reply:
I think we need this for the validators to run during the upgrade or else we run the risk of the bech32ibc module not being set up correctly. We wouldn't need this if we were running a genesis upgrade if I understand things correctly.

Reply:
oh I just meant the comments, wasn't sure if it as a todo or nod.